### DISCUSSION TITLE
Sync/Reactive atoms

### DISCUSSION THREAD

**Discussion by Odonno** [pmndrs/jotai#1722] [Ideas] [answered]
Hi @dai-shi,

Is it possible to create a new kind of atom called sync atom or reactive atom which can apply `set` function on each `get` invocation? Example below:

```ts
const namesAtom = atom([]);
const descriptionsAtom = atom([]);

const syncDescriptionsAtom = atom(
  (get, set) => {
    const names = get(namesAtom);
    const descriptions = get(descriptionsAtom);

    if (names.length === descriptions.length) {
      return;
    }

    const newDescriptions = [...descriptions];

    while (newDescriptions.length > names.length) {
      newDescriptions .pop();
    }
    while (newDescriptions.length < names.length) {
      newDescriptions .push("");
    }

    set(descriptionsAtom, newDescriptions);
  }
);
```

In this example, the descriptions will always be synced based on the value of names and the current value of descriptions.

As you can see, the atom function only has one parameter function which contains the `get` and the `set` functions. So, I believe it's possible to detect this new kind of atom.

Thank you.

**Comment by dai-shi** [comment, marked as answer]
No, the read function of atom must be "pure", no side effects like writing to atoms. You should prefer deriving atoms. If descriptionsAtom can simply be derived from namesAtom, it's the best. If it requires another state, create a new atom, and derive from two atoms.
